{
 "cells": [
  {
   "cell_type": "raw",
   "id": "f5f2a743",
   "metadata": {},
   "source": [
    "---\n",
    "output-file: 13-gut_microbiome.html\n",
    "title: 13-gut_microbiome\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9f2fbc1",
   "metadata": {},
   "source": [
    "# 13 - gut microbiome\n",
    "\n",
    "Metagenomics is the study of genetic material from environmental samples, including microbial communities. It involves sequencing the DNA of all microorganisms in the sample, rather than isolating individual organisms. Metagenomics enables the identification and functional analysis of microorganisms in diverse environments, including soil, water, and the human body.\n",
    "\n",
    "This dataset maps out the human gut microbiota per participant via shotgun metagenomic sequencing given stool samples. It is then compared to known references of gut flora to measure prevalence of specific microbes.\n",
    "\n",
    "Gut microbiome metagenomics can be used to identify potential biomarkers of disease, develop personalized treatment strategies, and better understand the complex relationship between the gut microbiota and human physiology. It has already been providing insights into the role of gut microbes in various diseases such as obesity, diabetes and IBD.\n",
    "\n",
    "To measure the genetic makeup of the human gut flora given stool samples via metagenomics, the following steps done:\n",
    "\n",
    "1. Collection of stool sample: For each visit, a stool sample is collected from the individual and stored appropriately to preserve the microbial community.\n",
    "2. DNA extraction: DNA is extracted from the stool sample using specialized techniques to isolate the microbial DNA from other materials present in the sample.\n",
    "3. DNA fragmentation and sequencing: The extracted DNA is then fragmented into small pieces and sequenced using high-throughput sequencing technologies.\n",
    "4. Quality control: The resulting raw sequencing data is then pre-processed, removing low-quality reads and artifacts of the sequencing methodology.\n",
    "5. Taxonomic classification: The processed sequencing data is then compared to databases of known microbial sequences to identify and classify the microbial species presence and their respective abundances in the sample.\n",
    "\n",
    "The information is stored in multiple parquet files:\n",
    "\n",
    "- `metadata`: Sequencing and QC statistics.\n",
    "- `urs`: Segal Lab relative abundance.\n",
    "- `metaphlan_*`: 8 tables with MetaPhlAn 4 relative abundances, separated by taxonomic levels.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "126bd297",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-29T16:12:51.896445Z",
     "iopub.status.busy": "2023-04-29T16:12:51.896104Z",
     "iopub.status.idle": "2023-04-29T16:12:52.184982Z",
     "shell.execute_reply": "2023-04-29T16:12:52.184213Z"
    }
   },
   "outputs": [],
   "source": [
    "#| hide\n",
    "import pandas as pd\n",
    "pd.set_option('display.max_rows', 500)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3e604258",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-29T16:12:52.187875Z",
     "iopub.status.busy": "2023-04-29T16:12:52.187637Z",
     "iopub.status.idle": "2023-04-29T16:12:53.214247Z",
     "shell.execute_reply": "2023-04-29T16:12:53.213640Z"
    }
   },
   "outputs": [],
   "source": [
    "from pheno_utils import PhenoLoader"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6c075acf",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-29T16:12:53.216823Z",
     "iopub.status.busy": "2023-04-29T16:12:53.216677Z",
     "iopub.status.idle": "2023-04-29T16:15:19.613498Z",
     "shell.execute_reply": "2023-04-29T16:15:19.612426Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DataLoader for gut_microbiome with\n",
       "9903 fields\n",
       "10 tables: ['metadata', 'urs', 'metaphlan_kingdom', 'metaphlan_phylum', 'metaphlan_class', 'metaphlan_order', 'metaphlan_family', 'metaphlan_genus', 'metaphlan_species', 'metaphlan_strain']"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dl = PhenoLoader('gut_microbiome', age_sex_dataset=None)\n",
    "dl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6034f0ff",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-29T16:15:19.617821Z",
     "iopub.status.busy": "2023-04-29T16:15:19.616287Z",
     "iopub.status.idle": "2023-04-29T16:15:19.634049Z",
     "shell.execute_reply": "2023-04-29T16:15:19.633586Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>field_string</th>\n",
       "      <th>description_string</th>\n",
       "      <th>parent_dataframe</th>\n",
       "      <th>relative_location</th>\n",
       "      <th>value_type</th>\n",
       "      <th>units</th>\n",
       "      <th>sampling_rate</th>\n",
       "      <th>item_type</th>\n",
       "      <th>array</th>\n",
       "      <th>cohorts</th>\n",
       "      <th>data_type</th>\n",
       "      <th>debut</th>\n",
       "      <th>pandas_dtype</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tabular_field_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>collection_timestamp</th>\n",
       "      <td>Sampled Timestamp</td>\n",
       "      <td>Time sample was given</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metadata.parquet</td>\n",
       "      <td>Date</td>\n",
       "      <td>Time</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>datetime64[ns]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>collection_date</th>\n",
       "      <td>Sampled Date</td>\n",
       "      <td>Date sample was given</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metadata.parquet</td>\n",
       "      <td>Date</td>\n",
       "      <td>Time</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>datetime64[ns]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SampleName</th>\n",
       "      <td>Sample Name</td>\n",
       "      <td>Sample Name</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metadata.parquet</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>raw_read_count</th>\n",
       "      <td>Raw Reads Count</td>\n",
       "      <td>#Reads in the original FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metadata.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>int64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed_read_count</th>\n",
       "      <td>Post QC Reads Count</td>\n",
       "      <td>#Reads in the Post QC FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metadata.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>int64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k__Bacteria|p__Verrucomicrobia|c__Verrucomicrobiae|o__Verrucomicrobiales|f__Akkermansiaceae|g__GGB6529|s__GGB6529_SGB9222|t__SGB9222</th>\n",
       "      <td>SGB9222</td>\n",
       "      <td>SGB9222</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metaphlan_strain.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Percent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k__Eukaryota|p__Ascomycota|c__Saccharomycetes|o__Saccharomycetales|f__Saccharomycetaceae|g__Saccharomyces|s__Saccharomyces_cerevisiae|t__EUK4932</th>\n",
       "      <td>EUK4932</td>\n",
       "      <td>EUK4932</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metaphlan_strain.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Percent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k__Eukaryota|p__Eukaryota_unclassified|c__Eukaryota_unclassified|o__Eukaryota_unclassified|f__Entamoebidae|g__Entamoeba|s__Entamoeba_dispar|t__EUK46681</th>\n",
       "      <td>EUK46681</td>\n",
       "      <td>EUK46681</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metaphlan_strain.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Percent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k__Eukaryota|p__Eukaryota_unclassified|c__Eukaryota_unclassified|o__Eukaryota_unclassified|f__Eukaryota_unclassified|g__Blastocystis|s__Blastocystis_sp_subtype_1|t__EUK944036</th>\n",
       "      <td>EUK944036</td>\n",
       "      <td>EUK944036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metaphlan_strain.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Percent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k__Eukaryota|p__Eukaryota_unclassified|c__Eukaryota_unclassified|o__Eukaryota_unclassified|f__Hexamitidae|g__Giardia|s__Giardia_intestinalis|t__EUK5741</th>\n",
       "      <td>EUK5741</td>\n",
       "      <td>EUK5741</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metaphlan_strain.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Percent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9903 rows Ã— 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                           field_string  \\\n",
       "tabular_field_name                                                        \n",
       "collection_timestamp                                  Sampled Timestamp   \n",
       "collection_date                                            Sampled Date   \n",
       "SampleName                                                  Sample Name   \n",
       "raw_read_count                                          Raw Reads Count   \n",
       "trimmed_read_count                                  Post QC Reads Count   \n",
       "...                                                                 ...   \n",
       "k__Bacteria|p__Verrucomicrobia|c__Verrucomicrob...              SGB9222   \n",
       "k__Eukaryota|p__Ascomycota|c__Saccharomycetes|o...              EUK4932   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...             EUK46681   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...            EUK944036   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...              EUK5741   \n",
       "\n",
       "                                                              description_string  \\\n",
       "tabular_field_name                                                                 \n",
       "collection_timestamp                                       Time sample was given   \n",
       "collection_date                                            Date sample was given   \n",
       "SampleName                                                           Sample Name   \n",
       "raw_read_count                                      #Reads in the original FASTQ   \n",
       "trimmed_read_count                                   #Reads in the Post QC FASTQ   \n",
       "...                                                                          ...   \n",
       "k__Bacteria|p__Verrucomicrobia|c__Verrucomicrob...                       SGB9222   \n",
       "k__Eukaryota|p__Ascomycota|c__Saccharomycetes|o...                       EUK4932   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...                      EUK46681   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...                     EUK944036   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...                       EUK5741   \n",
       "\n",
       "                                                    parent_dataframe  \\\n",
       "tabular_field_name                                                     \n",
       "collection_timestamp                                             NaN   \n",
       "collection_date                                                  NaN   \n",
       "SampleName                                                       NaN   \n",
       "raw_read_count                                                   NaN   \n",
       "trimmed_read_count                                               NaN   \n",
       "...                                                              ...   \n",
       "k__Bacteria|p__Verrucomicrobia|c__Verrucomicrob...               NaN   \n",
       "k__Eukaryota|p__Ascomycota|c__Saccharomycetes|o...               NaN   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...               NaN   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...               NaN   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...               NaN   \n",
       "\n",
       "                                                           relative_location  \\\n",
       "tabular_field_name                                                             \n",
       "collection_timestamp                                        metadata.parquet   \n",
       "collection_date                                             metadata.parquet   \n",
       "SampleName                                                  metadata.parquet   \n",
       "raw_read_count                                              metadata.parquet   \n",
       "trimmed_read_count                                          metadata.parquet   \n",
       "...                                                                      ...   \n",
       "k__Bacteria|p__Verrucomicrobia|c__Verrucomicrob...  metaphlan_strain.parquet   \n",
       "k__Eukaryota|p__Ascomycota|c__Saccharomycetes|o...  metaphlan_strain.parquet   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...  metaphlan_strain.parquet   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...  metaphlan_strain.parquet   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...  metaphlan_strain.parquet   \n",
       "\n",
       "                                                    value_type    units  \\\n",
       "tabular_field_name                                                        \n",
       "collection_timestamp                                      Date     Time   \n",
       "collection_date                                           Date     Time   \n",
       "SampleName                                                Text      NaN   \n",
       "raw_read_count                                         Integer      NaN   \n",
       "trimmed_read_count                                     Integer      NaN   \n",
       "...                                                        ...      ...   \n",
       "k__Bacteria|p__Verrucomicrobia|c__Verrucomicrob...  Continuous  Percent   \n",
       "k__Eukaryota|p__Ascomycota|c__Saccharomycetes|o...  Continuous  Percent   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...  Continuous  Percent   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...  Continuous  Percent   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...  Continuous  Percent   \n",
       "\n",
       "                                                    sampling_rate item_type  \\\n",
       "tabular_field_name                                                            \n",
       "collection_timestamp                                          NaN      Data   \n",
       "collection_date                                               NaN      Data   \n",
       "SampleName                                                    NaN      Data   \n",
       "raw_read_count                                                NaN      Data   \n",
       "trimmed_read_count                                            NaN      Data   \n",
       "...                                                           ...       ...   \n",
       "k__Bacteria|p__Verrucomicrobia|c__Verrucomicrob...            NaN      Data   \n",
       "k__Eukaryota|p__Ascomycota|c__Saccharomycetes|o...            NaN      Data   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...            NaN      Data   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...            NaN      Data   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...            NaN      Data   \n",
       "\n",
       "                                                     array cohorts data_type  \\\n",
       "tabular_field_name                                                             \n",
       "collection_timestamp                                Single     10K  Metadata   \n",
       "collection_date                                     Single     10K  Metadata   \n",
       "SampleName                                          Single     10K  Metadata   \n",
       "raw_read_count                                      Single     10K  Metadata   \n",
       "trimmed_read_count                                  Single     10K  Metadata   \n",
       "...                                                    ...     ...       ...   \n",
       "k__Bacteria|p__Verrucomicrobia|c__Verrucomicrob...  single     10K   Tabular   \n",
       "k__Eukaryota|p__Ascomycota|c__Saccharomycetes|o...  single     10K   Tabular   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...  single     10K   Tabular   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...  single     10K   Tabular   \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...  single     10K   Tabular   \n",
       "\n",
       "                                                         debut    pandas_dtype  \n",
       "tabular_field_name                                                              \n",
       "collection_timestamp                                2019-02-14  datetime64[ns]  \n",
       "collection_date                                     2019-02-14  datetime64[ns]  \n",
       "SampleName                                          2019-02-14          object  \n",
       "raw_read_count                                      2019-02-14           int64  \n",
       "trimmed_read_count                                  2019-02-14           int64  \n",
       "...                                                        ...             ...  \n",
       "k__Bacteria|p__Verrucomicrobia|c__Verrucomicrob...  2019-02-14         float64  \n",
       "k__Eukaryota|p__Ascomycota|c__Saccharomycetes|o...  2019-02-14         float64  \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...  2019-02-14         float64  \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...  2019-02-14         float64  \n",
       "k__Eukaryota|p__Eukaryota_unclassified|c__Eukar...  2019-02-14         float64  \n",
       "\n",
       "[9903 rows x 13 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dl.dict"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
